release-1.0.0

Satturas opened the pull request:
он же исправляет большую часть замечаний из итерэйшн-2

Antipin-A-A committed:
Удалил compiler.xml из репозитория, добавил в .gitignore

Satturas committed:
связь вьюмодели экрана создания списка с базой данных

Antipin-A-A committed:
Добавил разметку в secondFragment, и SecondScreenViewModel() функционал не полный

Antipin-A-A committed:
Добавил разметку в secondFragment, и SecondScreenViewModel() функционал не полный

Satturas committed:
Merge pull request #54 from nadinedenisova/11-32-databaseфича-2-стыковка-с-viewmodel

связь вьюмодели экрана создания списка с базой данных

Satturas committed:
Merge branch 'dev_com_new' into dev_Aleksandr_Фича-5_верстка_экрана_покупок

Satturas committed:
Merge pull request #53 from nadinedenisova/dev_Aleksandr_Фича-5_верстка_экрана_покупок

Dev aleksandr фича 5 верстка экрана покупок

Satturas committed:
настройка коин, исправление ошибок в БД

Satturas committed:
Merge pull request #57 from nadinedenisova/8-41-первая-рабочая-сборкафича-1-koin

настройка коин, исправление ошибок в БД

vedmapishet committed:
навигация, начало сведения mainscreen

vedmapishet committed:
правка мейнрепозиториИмпл

vedmapishet committed:
правка мейнрепозиториИмпл, сведение вьихи и главного экрана

vedmapishet committed:
переделан ShoplistScreenRepositoryImpl

vedmapishet committed:
работа над стейтами мэйнскрина

vedmapishet committed:
работа над стейтами мэйнскрина

Satturas committed:
Merge pull request #59 from nadinedenisova/10-43-первая-рабочая-сборкафича-3-навигация

10 43 первая рабочая сборкафича 3 навигация

Antipin-A-A committed:
на проверкуб подключаю ShoplistScreenViewModel к secondFragment.kt, дорабботал первый экран MainScreen.kt, изменил навигацию в NavGraph.kt

Antipin-A-A committed:
Merge pull request #61 from nadinedenisova/dev_Aleksandr_Настройка_второй_экран

на проверкуб подключаю ShoplistScreenViewModel к secondFragment.kt, д…

vedmapishet committed:
поправлены репозитории импл

vedmapishet committed:
поправлена функция переименования списка

vedmapishet committed:
поправлена функция закрепления/открепления списка

vedmapishet committed:
поправлена функция удаления списка

vedmapishet committed:
поправлена функция дублирования списка

vedmapishet committed:
добавлен и настроен detect

vedmapishet committed:
работа над ошибками detect: разбиты функции obtainEvent, ошибки с новыми строками в конце файла

vedmapishet committed:
работа над ошибками detect: разбитие функций mainScreen

vedmapishet committed:
работа над ошибками detect: разбитие функций mainScreen

vedmapishet committed:
работа над ошибками detect: разбитие функций mainScreen

Satturas committed:
Merge pull request #63 from nadinedenisova/25-71-detectфича-1-подключение

25 71 detectфича 1 подключение

Antipin-A-A committed:
на проверку , во втором экране не сделано редактирование и drag and drop

Antipin-A-A committed:
Merge pull request #65 from nadinedenisova/dev_Aleksandr_второй_экран

на проверку , во втором экране не сделано редактирование и drag and drop

vedmapishet committed:
настройка и подписание релизной сборки

Antipin-A-A committed:
исправил ошибку в методе "Выделить все продукты" и на главном экране ошибки ывета в диалогах

Antipin-A-A committed:
исправил ошибку в методе "Выделить все продукты" и на главном экране ошибки ывета в диалогах

Antipin-A-A committed:
исправил ошибку в методе "Выделить все продукты"  если isSelectProducts = true и добавить товар, то isSelectProducts оставался true

Antipin-A-A committed:
Merge pull request #66 from nadinedenisova/dev_Aleksandr_Исправление-IsChrckrd

Dev aleksandr исправление is chrckrd

vedmapishet committed:
настройка и подписание релизной сборки v 1.0.1

Secur1ty-21 commented on app/src/main/java/com/example/my_shoplist_application/di/InteractorModule.kt:
```diff
@@ -0,0 +1,18 @@
+val interactorModule = module {
+
+    single<MainScreenInteractor> {
+        MainScreenInteractorImpl(get())
```
Лучше использовать именованные параметры, чтобы было понятно что передается

Secur1ty-21 commented on app/src/main/java/com/example/my_shoplist_application/common/navigation/NavGraph.kt:
```diff
@@ -0,0 +1,69 @@
+    val navController = rememberNavController()
+
+    NavHost(navController = navController, startDestination = "shoppingList") {
+        composable("shoppingList") {
+            /*ShoppingListScreen2*/ MainScreen(
+                onListClick = {listId ->
+                    navController.navigate("previewIngredients/$listId")
+                }
+            )
+        }
+
+        composable(
+            "previewIngredients/{listId}",
+            arguments = listOf(
+                navArgument("listId") { type = NavType.IntType },
+            )
+        ) { backStackEntry ->
```
Советую посмотреть новый формат того как можно строить навигацию -> https://developer.android.com/guide/navigation/design/type-safety#type-safe

Secur1ty-21 commented on app/src/main/java/com/example/my_shoplist_application/presentation/MainScreen.kt:
```diff
@@ -53,186 +50,141 @@ import androidx.compose.ui.Alignment
+fun MainScreen(
+    onListClick: (Int) -> Unit
+) {
+    val viewModel: MainScreenViewModel = koinViewModel()
```
Лучше перенести в конструктор чтобы можно было подменить на тестах

Secur1ty-21 commented on app/src/main/java/com/example/my_shoplist_application/presentation/MainScreen.kt:
```diff
@@ -53,186 +50,141 @@ import androidx.compose.ui.Alignment
+    onListClick: (Int) -> Unit,
+    onDismiss: () -> Unit,
+) {
+    var newListName by remember { mutableStateOf("") }
```
Состояние не сохраниться при смене конфигурации, лучше rememberSaveale или вынести в viewModel

Secur1ty-21 commented on app/src/main/java/com/example/my_shoplist_application/presentation/MainScreen.kt:
```diff
@@ -53,186 +50,141 @@ import androidx.compose.ui.Alignment
+            Button(
+                onClick = {
+                    if (newListName.isNotBlank()) {
+                        viewModel.obtainEvent(MainScreenEvent.Add(newListName))
+                        viewModel.obtainEvent(MainScreenEvent.OnCloseAddingWindow)
+                        viewModel.obtainEvent(MainScreenEvent.Default)
```
Лучше обьеденить в один event, при быстром вызове нескольких могут быть проблемы с обновлением State

Secur1ty-21 commented on app/src/main/java/com/example/my_shoplist_application/presentation/MainScreen.kt:
```diff
@@ -53,186 +50,141 @@ import androidx.compose.ui.Alignment
+
+@Composable
+private fun DialogAddNameList(
+    viewModel: MainScreenViewModel,
```
Можно убрать viewModel из параметров и делать вызов необходимых функций в onListClick в месте вызова DialogAddNameList

Secur1ty-21 commented on app/src/main/java/com/example/my_shoplist_application/presentation/ShoplistScreen.kt:
Хорошее разделение ui элементов на функции

Secur1ty-21 reviewed (commented):
Высылаю оценку в соответсвии с ТЗ
# Техническое задание
## Разработка мобильного приложения «Список покупок»
## Формат оценок
- (не реализовано/ бОльшая часть сделана / реализовано, но есть баг/ реализовано)
- (не реализовано/реализовано, но есть баг/ реализовано)
- (не реализовано / реализовано)
## Общие требования
1. Стек
   - UI: Compose + MVI (не соблюден)
   - DB: Room
   - DI: Koin
   - Network: Retrofit
2. Сохранение всех данных в бд +
3. (*) Поддержка темной / светлой темы: (2)
4. (*) Добавлен и настроен detekt и соблюдены правила (1, не все правила соблюдены):
  - тело функции меньше 50 строк
  - количество аргментов в функции меньше 6
  - тело класса меньше 350 строк
5. (*) Поддержка смены ориентации экрана (0)
6. (*) Поддержка маленьких экранов (Small phone) (0.5)
При добавлении более 7 продуктов кнопка добавить новый продукт становится недоступной
7. (*) Отдельная верстка для планшетов (Возможность видеть все списки покупок и редактировать их на одном экране) (0)
8. Отсутствие runtime exception (2)
9. Наличие багов (runtime exception + визуальные):
После создания списка, он не сразу отображает во всех списках, только после создания еще одного
Удаление списка покупок не отображает изменений на экране всех списков
Дублирование списка покупок вместо добавления нового списка заменяет списки над дублируемым
Дублирование списка покупок не дублирует товары в списке
При добавлении более 15 продуктов кнопка добавить новый продукт становится недоступной
  - Меньше 6( +1 балл)
10. Настроить рилизную сборку (proguard + minifyEnable) + и подписать приложение (https://developer.android.com/studio/publish/app-signing) - приложение не подписано

## Главный экран

1. Возможность перейти к созданию нового списка покупок (1)
2. Отображение всех созданных списков покупок (1)
3. Возможность переименовать/удалить список покупок (1)
4. При удалении списка покупок показать диалог с подтверждением действия (1)
5. (*) Вызов переименовывания/удаления списка через свайп элемента (2)
Свайп становится недоступным если провести до границы
6. (*) Возможность дублирования существующего списка (1)

## Экран редактирования/создания списка покупок

1. Возможность добавить товар с указанием количества и единицы измерения (опционально) (шт. , кг. , л. , мл. , г.) (2)
2. Редактирование / удаление товара (1)
Редактирование товара не реализовано
3. Возможность пометить товар как купленный (1)
4. Добавить контекстное меню с возможностью переименовать / удалить / очистить список(удаляет все купленные товары) товаров (2)
5. Возможность сортировки списка товаров по алфавиту (0)
6. (*) Вызов редактирования / удаления товара через свайп (3)
7. (*) При добавлении нового товара сохранять его имя в отдельной таблице в БД и, при следующем добавлении товара, делать всплывающие подсказки по имени товара с возможностью подставить имя товара по тапу на подсказку (5)
8. (*) Возможность ручной сортировки списка товаров через drag and drop (0)

Antipin-A-A committed:
перенес навигацию, добавил анимацию при переходе

Antipin-A-A committed:
Merge pull request #67 from nadinedenisova/dev_Aleksandr_Редактирование

перенес навигацию, добавил анимацию при переходе

Satturas committed:
правки после ревью:
- поправлен баг не отображения вновь добавленного списка
- поправлен баг отображения удаленного списка
- добавлены поименованные параметры в DI
- вьюмодели перенесены в конструкторы в экранах
- объединен в один ивент при добавлении нового списка
- переделана навигация (Александр)
